### PR TITLE
Feature/misc panic and coredump fixes

### PR DESCRIPTION
**Pull request by tim-nordell-nimbelink**
I believe one of the related issues is #1394 (I don't know if it's the same).  Namely when a core dump occurs it tries to extend the stack region to dump to match the tip from within the exception frame since the TCB may not have been updated to include the full stack region.  However if we're in the middle of a normal interrupt the exception frame points into the system stack instead of the task's stack.  As such it ends up extending the region to dump to be much further than it should.

I also added a couple of tests into the coredump test routine (but none of the expected values for the PC side), and an exclusion of grabbing the esp_ota_data when spi flash cache is disabled.

### CODE REVIEW

File: components/esp32/panic.c [hunk 1]
```diff
@@ -543,10 +543,14 @@ static void commonErrorHandler_dump(XtExcFrame *frame, int core_id)
     }
 
     panicPutStr("\r\nELF file SHA256: ");
-    char sha256_buf[65];
-    esp_ota_get_app_elf_sha256(sha256_buf, sizeof(sha256_buf));
-    panicPutStr(sha256_buf);
-    panicPutStr("\r\n");
+    if (spi_flash_cache_enabled()) {
```
[the remark this answers is not in the document]
Comment (tim-nordell-nimbelink) [reply components/esp32/panic.c]: Sorry, the "deeper look" comment was buried in the commit description (https://github.com/espressif/esp-idf/pull/3650/commits/34a83c6ae5b259da1e4d49d918472a5ace37bc82) for this change - the fact that the description it reads is within flash.  If I recall correctly, the data it reads back is of the form 0xbad00bad across the address space, so the sha256 value has that byte repeated across it.  It just looked weird when I was testing things.

File: components/espcoredump/test/test_core_dump.c [hunk 2]
```diff
@@ -96,10 +98,43 @@ void failed_assert_task(void *pvParameter)
...
+    printf("Task 'crash_from_no_flash_cache' start.\n");
+    while (1) {
+        vTaskDelay(1000 / portTICK_RATE_MS);
+        printf("Task 'crash_from_no_flash_cache' run.\n");
+        crash_from_ram_no_flash_cache();
+    }
+    fflush(stdout);
+}
+
+void starve_idle()
```
[the remark this answers is not in the document]
Comment (tim-nordell-nimbelink) [reply components/espcoredump/test/test_core_dump.c]: It helped see issues with the panic handler, for instance, of the elf sha256 sum being weird in this case.  You're probably right in that this probably more aptly belongs inside a test case for the panic handler, and not the coredump component.

In our local repository, however, I do have branch code within the coredump component dependent on if flash cache is enabled.  (I have a custom backend for the coredump routine which dumps the information into a filesystem that isn't supported within esp-idf - I've been poking at https://github.com/ARMmbed/littlefs.)  Hence why I came across some of the issues present in some of the wrapper code around running coredumps as I was debugging my coredump implementation and hitting some crashes within it.  (For instance, I had a crash in my coredump code when cache flash was disabled, in which I hit the bug fixed in #3661.  That was being tested by a test case similar to this, although, I have it implemented as a test within our console code instead of this test case specifically, but the test is the same.)
